assets/js/modules/tagmanager/datastore/existing-tag.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * modules/tagmanager data store: existing tag.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { getExistingTag } from '../../../util/tag';
+import { isValidContainerID } from '../util/validation';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+
+const { createRegistrySelector } = Data;
+
+const fetchGetExistingTagStore = createFetchStore( {
+	baseName: 'getExistingTag',
+	controlCallback: () => getExistingTag( 'tagmanager' ),
+	reducerCallback: ( state, existingTag ) => {
+		return {
+			...state,
+			existingTag,
+		};
+	},
+} );
+
+const fetchGetTagPermissionStore = createFetchStore( {
+	baseName: 'getTagPermission',
+	argsToParams: ( tag ) => {
+		invariant( isValidContainerID( tag ), 'a valid tag is required to for fetching and receiving permission.' );
+
+		return { tag };
+	},
+	controlCallback: ( { tag } ) => API.get( 'modules', 'tagmanager', 'tag-permission', { tag }, { useCache: false } ),
+	reducerCallback: ( state, { permission }, { tag } ) => {
+		return {
+			...state,
+			tagPermission: {
+				...state.tagPermission,
+				[ tag ]: permission,
```

Comment:
I agree that there is value in the association that comes back from this as it is useful for a subsequent containers request, but I'm not sure that storing the account ID with the permission itself is the right place for it.

If the intention is to store references to which container belongs to which account, that should probably have its own selector and state which maps `[ containerID ]: accountID`. It doesn't make sense to check tag permission to get a related accountID. Is there another reason why it should be stored here?

Reply:
Since the account ID is being detected as part of the permission check, it's stored here. I agree it feels a bit odd, but this is how we've been doing it in the other areas as well. I'd prefer to stick to this convention for now, also to get this in, and discuss via a new issue how we can improve that.